json_serialization_writer.go

```diff
@@ -609,6 +609,27 @@ func (w *JsonSerializationWriter) GetSerializedContent() ([]byte, error) {
 	return []byte(resultStr), nil
 }
 
+// WriteAnyValues an unknown value as a parameter.
+func (w *JsonSerializationWriter) WriteAnyValues(key string, value interface{}) error {
+
+	if value != nil {
+		body, err := json.Marshal(value)
```

Comment:
same comment about trying through scalar values before marshalling.

Reply:
Write any values is only called as the default  in `WriteAdditionalData` at this point all the types have failed to match.

Reply:
ah thanks for the clarification, late review on the phone, after a long flight, I didn't have all context